src/lib.rs

```diff
@@ -72,234 +78,143 @@ impl Plugin for BevyLocalCommandsPlugin {
-                let mut time = SystemTime::now();
-                loop {
-                    // TODO: FIXME: This blocks the thread, which is why we can't kill until there's some output
-                    // Tried using heim library to kill process but had dependency hell issues
-                    let option_result_line = lines.next();
-                    if let Some(result_line) = option_result_line {
-                        if let Ok(line) = &result_line {
-                            if let Ok(mut asc) = asc_moved.lock() {
-                                output_lines.push(line.clone());
-                                let time_elapsed = time.elapsed().unwrap_or_default();
-                                // Only submit buffered lines and check for kill command every second
-                                if time_elapsed > std::time::Duration::from_secs(1) {
-                                    time = SystemTime::now();
-                                    asc.output_lines = Some(output_lines.clone());
-                                    output_lines.clear();
-                                    // Check for kill command.
-                                    if asc.kill_requested {
-                                        if reader_handle.kill().is_ok() {
-                                            break;
-                                        } else {
-                                            error!("Failed to kill process PID {}", pid)
-                                        }
-                                    }
-                                }
-                            } else {
-                                error!(
-                                    "Failed to access active shell command for PID {}: {}",
-                                    pid, command_string
-                                );
-                                warn!("We have probably lost command output lines");
-                            }
-                            continue;
-                        } else {
-                            error!(
-                                "Command Exit Error (PID {}): {:?}",
-                                pid,
-                                result_line.as_ref().err()
-                            );
-                        }
-                    } else {
-                        info!("Command (PID {}) Completed: {}", pid, &command_string);
-                        result = true;
-                    }
-                    break;
-                }
-
-                (result, output_lines)
-            });
-            if let Ok(mut asc) = active_shell_command.lock() {
-                asc.task = Some(task);
-            } else {
-                error!("Failed to add new active shell command: {}", &cs)
+fn spawn_process(mut command: Command) -> ActiveProcess {
+    // Start running the process
+    let mut process = command.spawn().unwrap();
```

Comment:
I recognize there's still some hardening to be done, but for this PR can we handle the scenario where a invalid executable is provided instead of just panicking on the unwrap? I'm getting this output when program isn't valid:
```
thread 'Compute Task Pool (3)' panicked at src\lib.rs:184:39:
called `Result::unwrap()` on an `Err` value: Error { kind: NotFound, message: "program not found" }
```

Maybe spawn_process can return a Result instead? Open to other ideas.

Reply:
Good point, we should handle this.
Maybe having an event stream for errors where the user can react to this?

Reply:
@edouardpoitras Sorry for the delay, I now added an event for errors and use that when the process failed to start instead of crashing the app.
In future PRs we can expand this mechanism for other kinds of errors that can occur.

I also renamed most events from `command` to `process` in order to distinguish a bit from `std::process::Command`.
But if you don't like the new names I can also revert those changes.

Reply:
No worries, there's no rush here - I'm just happy to have some help.
No issues on the rename.
I added a few comments for review.
Thanks.